### ISSUE TITLE
RestClient.Get does not execute the code

### ISSUE THREAD

**Issue by DevingToshuk** [proyecto26/RestClient#166]
When I write
```c#
RestClient.Get<Data>("https://databasehere.com/" + index + ".json").Then(response => 
{
       Debug.Log(response._name);
});
```

No message in the console is printing. This is executed in the Start() function and the class is pre-uploaded in the data base, whit the use of RestClient.Put. It appears that after the response is obtained the code is not being executed, or maybe the response is not obtained at all.

Why is the Debug.Log(response._name) not being executed?

**Comment by jdnichollsc** [comment]
Try using Catch method and let me know 🙂

**Comment by DevingToshuk** [comment]
There is no exception being thrown, the problem is that nothing happens when I use the RestClient.Get. I have a class, the one I send and retrieve from the database. I'm using Firebase Realtime Database. Here is the class I'm trying to retrieve
```c#
public class Data
{
    public int _number;
    public string _string;

    public Data(int Number, string String)
    {
        _number = Number;
        _string = String;
    }
}
```
so when I'm trying to retrieve the class that was previously uploaded to the database, using RestClient.Put, I call a method from which I execute the RestClient.Get and this is where the problem occurs. Here is the method

```c#
public void GetData()
{
       RestClient.Get<Data>("https://databaseproviderhere.com/" + index + ".json").Then(response =>
       {
              Debug.Log(response._string);
       });
}
``` 
In the example above nothing happens, no message, no exception, absolutely nothing. I checked and the value is present at the database.


Later I ran another test in which I attempted to retrieve the _number.

```c#
public void GetData()
{
       RestClient.Get<Data>("https://databaseproviderhere.com/" + index + ".json").Then(response =>
       {
              retrievedNumber = response._number;
       });
}

public void DisplayData()
{
       Debug.Log(retrievedNumber);
}
```

The log showed that retrievedNumber was equal to Null. The DisplayData() method is being called after the GetData()

**Comment by jdnichollsc** [comment]
Can you attach a reproducible demo (repository) to debug your issue please?
Some useful resources:
- https://medium.com/@rotolonico/firebase-authentication-in-unity-with-rest-api-7fc8c1576178
- https://stackoverflow.com/questions/66510364/update-single-value-in-firebase-realtime-database-using-rest-api

**Comment by DevingToshuk** [comment]
Figured it out) Took me a while, after reading the https://medium.com/@rotolonico/firebase-authentication-in-unity-with-rest-api-7fc8c1576178 I saw the Full Serializer library. After looking trough the code examples I saw that the class that was being retrieved was serializable. So then I tried

```c#
[System.Serializable]
public class Data
{
    public int _number;
    public string _string;

    public Data(int Number, string String)
    {
        _number = Number;
        _string = String;
    }
}
```

and it worked, the RestClient.Get line worked and did what it was supposed to! Thank you for the links you provided) So if anyone is having trouble with RestClient.Get not working just make sure the class you are trying to retrieve is serializable.
